packages/dd-trace/src/plugins/database.js

```diff
@@ -1,9 +1,63 @@
 'use strict'
 
 const StoragePlugin = require('./storage')
+const TextMapPropagator = require('../opentracing/propagation/text_map.js')
+const Config = require('../../src/config')
+const config = new Config()
+const textMap = new TextMapPropagator(config)
+const carrier = {}
 
 class DatabasePlugin extends StoragePlugin {
   static get operation () { return 'query' }
+
+  constructor (...args) {
+    super(...args)
+    this.serviceTags = {
+      dddbs: '',
+      encodedDddbs: '',
+      dde: '',
+      encodedDde: '',
+      ddps: '',
+      encodedDdps: '',
+      ddpv: '',
+      encodedDdpv: ''
+    }
+  }
+
+  createDBMPropagationCommentService () {
+    if (this.serviceTags.dddbs !== this.config.service) {
+      this.serviceTags.dddbs = this.config.service
+      this.serviceTags.encodedDddbs = encodeURIComponent(this.config.service)
+    }
+    if (this.serviceTags.dde !== this.tracer._env) {
+      this.serviceTags.dde = this.tracer._env
+      this.serviceTags.encodedDde = encodeURIComponent(this.tracer._env)
+    }
+    if (this.serviceTags.ddps !== this.tracer._service) {
+      this.serviceTags.ddps = this.tracer._service
+      this.serviceTags.encodedDdps = encodeURIComponent(this.tracer._service)
+    }
+    if (this.serviceTags.ddpv !== this.tracer._version) {
+      this.serviceTags.ddpv = this.tracer._version
+      this.serviceTags.encodedDdpv = encodeURIComponent(this.tracer._version)
+    }
+
+    return `dddbs='${this.serviceTags.encodedDddbs}',dde='${this.serviceTags.encodedDde}',` +
+    `ddps='${this.serviceTags.encodedDdps}',ddpv='${this.serviceTags.encodedDdpv}'`
+  }
+
+  injectDbmQuery (query) {
+    if (this.config.dbmPropagationMode === 'service') {
+      const servicePropagation = this.createDBMPropagationCommentService()
+      return `/*${servicePropagation}*/ ${query}`
+    } else if (this.config.dbmPropagationMode === 'full') {
+      config.experimental.traceparent = true
```

Comment:
The config should never be changed from a plugin.

Reply:
I removed this since I am no longer calling the text map function.